src/rust/ensogl/src/math/circle_segment.rs

```diff
@@ -0,0 +1,55 @@
+//! Provides functionality related to circle segments.
+use crate::math::topology::unit::{Angle, Radians};
+
+
+
+// =====================
+// === CircleSegment ===
+// =====================
+
+/// Implements computations related to circle segments.
+/// For details and background on the formulas used, see
+/// https://en.wikipedia.org/wiki/Circular_segment
+#[derive(Clone,Copy,Debug,PartialEq)]
+pub struct CircleSegment{
+    radius: f32,
+    angle : Angle<Radians>,
+}
+
+impl CircleSegment{
+
+    /// Constructor.
+    pub fn new(radius:f32, angle:Angle<Radians>) -> Self{
+        CircleSegment{radius, angle}
+    }
+
+    /// The length of the direct line between the segment end points.
+    pub fn chord_length(self) -> f32{
+        let r     = self.radius;
```

Comment:
We are not using 1-letter variables if we can. This should be named `radius` instead.